src/App.tsx

```diff
@@ -1,24 +1,119 @@
+import { USER_ID } from './utils/fetchClient';
+import { useError } from './utils/useError';
+import { LoadStatusContext } from './utils/LoadStatusContext';
+
+import { initData } from './constants/initData';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(initData.todos);
+  const [loadingStatus, setLoadingStatus]
+    = useState<number[]>(initData.loadingState);
+  const [filter, setFilter] = useState<Filter>(initData.filter);
+  const [customError, setError]
+    = useError(initData.customError);
+  const [activeLeft, setActiveLeft]
+    = useState<number>(initData.activeLeft);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(initData.tempTodo);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(todosFromServer => {
+        setTodos(todosFromServer);
+        setActiveLeft(todosFromServer.filter(todo => !todo.completed).length);
+      })
+      .catch(() => {
+        setError(CustomError.Update, 3000);
+      });
+  }, []);
+
+  useEffect(() => {
+    setActiveLeft(todos.filter(todo => !todo.completed).length);
+    setError(CustomError.NoError);
+  }, [todos]);
+
+  const filteredTodos = useMemo(() => {
+    switch (filter) {
+      case Filter.All: {
+        return todos;
+      }
+
+      case Filter.Active: {
+        return todos.filter(todo => !todo.completed);
+      }
+
+      case Filter.Completed: {
+        return todos.filter(todo => todo.completed);
+      }
+
+      default: {
+        setError(CustomError.Update);
+
+        return todos;
+      }
+    }
+  },
+  [todos, filter]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <LoadStatusContext.Provider value={{ loadingStatus, setLoadingStatus }}>
+      <div className="todoapp">
+        <h1 className="todoapp__title">todos</h1>
+
+        <div className="todoapp__content">
+          <Header
+            todos={todos}
+            setTodos={setTodos}
+            tempTodo={tempTodo}
+            setTempTodo={setTempTodo}
+            activeLeft={activeLeft}
+            setError={setError}
+          />
+
+          <ListOfTodos
+            todos={filteredTodos}
+            setTodos={setTodos}
+            tempTodo={tempTodo}
+            setError={setError}
+          />
+
+          {!!todos.length && (
+            <FooterMenu
+              todos={filteredTodos}
+              setTodos={setTodos}
+              filter={filter}
+              setFilter={setFilter}
+              activeLeft={activeLeft}
+              setError={setError}
+            />
+          )}
+        </div>
+
+        {customError && (
```

Comment:
As I see it wouldn't be rendered anyway, so it looks like useless check
![image](https://user-images.githubusercontent.com/122989585/223819837-e08debf9-b4da-4198-9f0b-ac277adcffd5.png)